### ISSUE TITLE
unclear process for generating data/ actual attack not implemented?

### ISSUE THREAD

**Issue by paullintilhac** [MultimodalMI/Multimodal-membership-inference#3]
Hello, thank you for posting this repository. I was able to get the training code to run after fixing some bugs, but it isn't clear to me how to make the different datasets for your main experiment. You say to reference the paper, but the code in your repository does not match that description. The code shows that 

In your paper, you say that the shadow model is used to calculate the threshold used for membership inference, and that you achieve high accuracy of inference. However I don't see any code that actually corresponds to these tests. First of all, the limited code that is in the repository shows this:
```
f8k_images = random.sample(f8k_image_path, 1000)
f8k_images_out = random.sample(f8k_image_test_path, 1000)
f8k_images_shadow = random.sample(f8k_image_path, 1000)
```
this appears to correspond to the "restricted knowledge" scenario, since the shadow models are coming from a different distribution. However the data is 3x less. And there is also no separate sampling process for the shadow non-member data. It appears that the shadow non-member data is just the same as the target model's non member data. That does not match the description, so this experiment isn't the correct one. Which one is the correct process? do we sample the shadow non-member data from the "other" dataset, as in the description, or simply use the target model non-member data, as in the code sample here?

Also, after you calculate the similarity scores for the target and shadow model on all relevant data points, there is no more corresponding to an actual attack, or calculation of performance, etc. It looks like it just saves some text files with the similarity scores and then finishes without any attack. Is there some code missing? It would be great to be able to actually reproduce this, but the current code can't be used to reproduce as-is.

Are there any plans to make this reproducible in the future?

Thank you,

Paul
